block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala

```diff
@@ -77,8 +77,20 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
       finalizedBlocksSet.contains(blockHash).pure[F]
 
     override def find(truncatedHash: String): F[Option[BlockHash]] = Sync[F].delay {
-      val truncatedByteString = truncatedHash.unsafeHexToByteString
-      dagSet.find(hash => hash.startsWith(truncatedByteString))
+      if (truncatedHash.length % 2 == 0) {
+        val truncatedByteString = truncatedHash.unsafeHexToByteString
+        dagSet.find(hash => hash.startsWith(truncatedByteString))
+      } else {
+        // if truncatedHash is odd length string we cannot convert it to ByteString with 8 bit resolution
+        // because each symbol has 4 bit resolution. Need to make a string of even length by removing the last symbol,
+        // then find all the matching hashes and choose one that matches the full truncatedHash string
+        val truncatedByteString = truncatedHash.dropRight(1).unsafeHexToByteString
+        dagSet
+          .filter(hash => hash.startsWith(truncatedByteString))
+          .find(
+            hash => Base16.encode(hash.toByteArray).startsWith(truncatedHash)
```

Comment:
You can use extension function on ByteString also created by @zsluedem in PR [#3517 (ByteStringSyntax)](https://github.com/rchain/rchain/pull/3517/files#diff-b467c85b5271a3e0eb2832ed6e1d5a9c6764ef3f65b5e8cfd9a0c7a01a57a0b8R35).
Together with shorthand for argument (`_`) in anonymous functions it can be one liner.
```scala
dagSet
  .filter(_.startsWith(truncatedByteString))
  .find(_.toHexString.startsWith(truncatedHash))
```

Nice solution to resolve incomplete hex problem. :+1: